frontend/pages/topics/index.tsx

```diff
@@ -1,40 +1,57 @@
 import Layout from "@/components/Layout";
 import PageHeader from "@/components/PageHeader";
-import { Topic } from "@/types/modules";
-import Button from "@material-ui/core/Button";
+import { ModuleUnion, Topic } from "@/types/modules";
+import { Button, Divider } from "@material-ui/core";
 import Container from "@material-ui/core/Container";
 import axios from "axios";
 import { GetServerSideProps } from "next";
 import Link from "next/link";
-import { FC } from "react";
+import React, { FC } from "react";
 
 interface TopicsProps {
   topicsData: {
-    topics: Pick<Topic, "name" | "slug">[];
+    topics: Pick<Topic, "name" | "slug" | "model">[];
   };
+  module: ModuleUnion;
+  className?: string;
 }
 
 const Topics: FC<TopicsProps> = ({ topicsData }: TopicsProps) => {
   const topics = topicsData.topics || [];
+  const topicsByLetters: Record<string, typeof topics> = {};
+  topics.map((topic) => {
+    const firstLetter = topic.name[0].toUpperCase() || " ";
+    topicsByLetters[firstLetter] = topicsByLetters[firstLetter] || []; //create array if not created already
```

Comment:
I think we can remove the `|| " "` here.

Also nit: can we include a space after the `//` in the comment?